beekeeper-api/src/main/java/com/expediagroup/beekeeper/api/error/BeekeeperExceptionHandler.java

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright (C) 2019-2024 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.expediagroup.beekeeper.api.error;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.springframework.data.mapping.PropertyReferenceException;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+import java.time.LocalDateTime;
+
+@RestControllerAdvice
+public class BeekeeperExceptionHandler {
+
+  /**
+   * Handles invalid sort parameters.
+   *
+   * @param ex      the exception thrown when an invalid property is referenced
+   * @param request the HTTP request
+   * @return a ResponseEntity containing the error response
+   */
+
+  @ExceptionHandler(PropertyReferenceException.class)
+  public ResponseEntity<ErrorResponse> handlePropertyReferenceException(
+      PropertyReferenceException ex, HttpServletRequest request) {
```

Comment:
Better to use full name for readability
```suggestion
      PropertyReferenceException exception, HttpServletRequest request) {
```